proto/internal/uber/cadence/admin/v1/service.proto

```diff
@@ -0,0 +1,286 @@
+}
+
+message GetWorkflowExecutionRawHistoryV2Response {
+  bytes next_page_token = 1;
+  repeated api.v1.DataBlob history_batches = 2;
+  shared.v1.VersionHistory version_history = 3;
+}
+
+message GetReplicationMessagesRequest {
+  repeated shared.v1.ReplicationToken tokens = 1;
+  string cluster_name = 2;
+}
+
+message GetReplicationMessagesResponse {
+  map<int32, shared.v1.ReplicationMessages> shard_messages = 1;
+}
+
+message GetDLQReplicationMessagesRequest {
+  repeated shared.v1.ReplicationTaskInfo task_infos = 1;
+}
+
+message GetDLQReplicationMessagesResponse {
+  repeated shared.v1.ReplicationTask replication_tasks = 1;
+}
+
+message GetDomainReplicationMessagesRequest {
+  // last_retrieved_message_id is where the next fetch should begin with.
+  int64 last_retrieved_message_id = 1;
+  // last_processed_message_id is the last messageId that is processed on the passive side.
+  // This can be different than lastRetrievedMessageId if passive side supports prefetching messages.
+  int64 last_processed_message_id = 2;
+  // cluster_name is the name of the pulling cluster.
+  string cluster_name = 3;
+}
+
+message GetDomainReplicationMessagesResponse {
+  shared.v1.ReplicationMessages messages = 1;
+}
+
+// ReapplyEventsRequest is the request for reapply events API.
+message ReapplyEventsRequest {
+  string domain = 1;
+  api.v1.WorkflowExecution workflow_execution = 2;
+  api.v1.DataBlob events = 3;
+}
+
+message ReapplyEventsResponse {
+}
+
+message AddSearchAttributeRequest {
+  map<string, api.v1.IndexedValueType> search_attribute = 1;
+  string security_token = 2;
+}
+
+message AddSearchAttributeResponse {
+}
+
+message DescribeClusterRequest {
+}
+
+message DescribeClusterResponse {
+  api.v1.SupportedClientVersions supported_client_versions = 1;
+  shared.v1.MembershipInfo membership_info = 2;
+}
+
+message ReadDLQMessagesRequest {
+  shared.v1.DLQType type = 1;
+  int32 shard_id = 2;
+  string source_cluster = 3;
+  int64 inclusive_end_message_id = 4;
+  int32 page_size = 5;
+  bytes next_page_token = 6;
+}
+
+message ReadDLQMessagesResponse {
+  shared.v1.DLQType type = 1;
+  repeated shared.v1.ReplicationTask replication_tasks = 2;
+  bytes next_page_token = 3;
+}
+
+message PurgeDLQMessagesRequest {
+  shared.v1.DLQType type = 1;
+  int32 shard_id = 2;
+  string source_cluster = 3;
+  int64 inclusive_end_message_id = 4;
+}
+
+message PurgeDLQMessagesResponse {
+}
+
+message MergeDLQMessagesRequest {
+  shared.v1.DLQType type = 1;
+  int32 shard_id = 2;
+  string source_cluster = 3;
+  int64 inclusive_end_message_id = 4;
+  int32 page_size = 5;
+  bytes next_page_token = 6;
+}
+
+message MergeDLQMessagesResponse {
+  bytes next_page_token = 1;
+}
+
+message RefreshWorkflowTasksRequest {
+  string domain = 1;
+  api.v1.WorkflowExecution workflow_execution = 2;
+}
+
+message RefreshWorkflowTasksResponse {
+}
+
+message ResendReplicationTasksRequest {
+  string domain_id = 1;
+  string workflow_id = 2;
+  string run_id = 3;
```

Comment:
Is there a uuid type in proto we should consider using? Or is string the standard way to represent uuids in proto?

Reply:
To my knowledge there is no default UUID type for proto, and [this](https://stackoverflow.com/questions/36344826/how-do-i-represent-a-uuid-in-a-protobuf-message) confirms that it best represented with either string or []byte. Given we already use strings everywhere, I would say we keep them here as well.

Reply:
yea, personally I think string is best for uuids.  it's dramatically easier to deal with when things fall apart than a bunch of bytes.  if representing UUIDs as strings vs bytes is proving to be a performance bottleneck, we have a truly awesomely optimized system - otherwise, optimizing for humans/failure modes seems best.